resources/fetchers/ecr_fetcher.go

```diff
@@ -101,15 +121,28 @@ func (f *ECRFetcher) describePodImagesRepositories(ctx context.Context, podsList
 			// Takes only aws images
 			regexMatcher := describer.FilterRegex.FindStringSubmatch(image)
 			if regexMatcher != nil {
-				repository := regexMatcher[1]
-				repositories = append(repositories, repository)
+				repository := regexMatcher[describer.ImageRegexIndex]
+				region := describer.ExtractRegion(describer, image)
+				reposByRegion[region] = append(reposByRegion[region], repository)
 			}
 		}
 	}
+	return reposByRegion
+}
+
+func ExtractRegionFromEcrImage(describer PodDescriber, image string) string {
+	regexMatcher := describer.FilterRegex.FindStringSubmatch(image)
+	if regexMatcher != nil {
+		repository := regexMatcher[EcrRegionRegexGroup]
+		return repository
+	}
 
-	f.log.Debugf("sending pods to ecrProviders: %v", repositories)
+	return ""
+}
 
-	return describer.Provider.DescribeRepositories(ctx, repositories)
+// ExtractRegionFromPublicEcrImage TODO Ofir - https://github.com/elastic/security-team/issues/4035
```

Comment:
the todo is not informative and the issue doesn't really explain it, please either remove this empty function or explain what's going on here in the ticket. Personally I'm against todos, especially if there is already a ticket that will track it.

Reply:
I added this code for test purposes, so I can test the public ECR as well.

I changed the TODO to be a bit more informative, if any other change is needed I'll be happy to do it